### ISSUE TITLE
Handling dependency updates

### ISSUE THREAD

**Issue by gilbitron** [coenjacobs/wordpress-composer-installer#5]
This is a great start to a solution to a big problem in WordPress at the moment. However one area not addressed at the moment is how to handle updates to dependencies. A simple solution is to setup a WP cron to run `composer update` twice daily (or however often) as I have done (crudely) for my own blog https://github.com/gilbitron/Blog/blob/master/public/content/mu-plugins/composer-updater.php. However this doesn't take into account breaking changes. Any thoughts on how to tackle this?

**Comment by coenjacobs** [comment]
Yep, valid point. I haven't thought about this much, as it would do most of this during plugin update or activation. Maybe even after deactivating one, but not sure, as the same dependencies would be compatible with the active plugins still.

Automated updates isn't really a focus at this point. This will probably lead to some issues, like putting the site in maintenance mode while it runs? Also, it's a very expensive process to run, so I'd rather have that run manually, seeing there are plenty of shared/shit hosted sites out there. :smile:

I do want to have some ways of updating the dependencies. Maybe a button somewhere in the Updates screen? Breaking changes will always be a problem, but nothing more explosive than they are currently in the WordPress plugins repository.

More ideas are always welcome.

**Comment by gilbitron** [comment]
Doing dependency updates during plugin install/update could work. I agree having a way to manually trigger the update probably makes sense. Do you think it might be worth having another tab inside the plugins page UI for "Dependencies" showing a list of installed dependencies and their versions? Maybe an update button could go on that screen?

**Comment by coenjacobs** [comment]
You mean one of these, right?

![screen shot 2015-07-08 at 08 01 58](https://cloud.githubusercontent.com/assets/245703/8564083/a6a69ee8-2547-11e5-9b9f-8855b8fbad08.png)

Or maybe we should add a new submenu under 'Plugins' in the main admin menu. Haven't really decided there yet, but I really like the idea of having a separate page/tab for it. That would be the perfect place for the manual update button. Maybe one day it won't be just dependencies for plugins though, but I guess we can move the screen then.

**Comment by gilbitron** [comment]
> You mean one of these, right?

Yes I do. Good point about themes. We might need to find a better location for a page like this but to start with putting it under plugins would work.
